File: Algorithms/405/toHex.java
class Solution {
    public String toHex(int num) {
        if (num == 0) {
            return "0";
        }
        char[] hex = "0123456789abcdef".toCharArray();
        StringBuilder sb = new StringBuilder();
        while (num != 0 && sb.length() < 8) {
            sb.append(hex[num & 0xf]);
            num >>= 4;
        }
        return sb.reverse().toString();
    }
}

File: Algorithms/405/toHex.py
class Solution:
    def toHex(self, num: int) -> str:
        if num == 0:
            return '0'
        if num < 0:
            num += 0xffffffff + 1
        chars = []
        hexchars = '0123456789abcdef'
        while num:
            num, r = divmod(num, 16)
            chars.append(hexchars[r])
        return ''.join(reversed(chars))
